resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/CircuitBreaker.java

```diff
@@ -556,13 +709,32 @@ static CheckedRunnable decorateCheckedRunnable(CircuitBreaker circuitBreaker, Ch
      * @return a consumer which is decorated by a CircuitBreaker.
      */
     static <T> Consumer<T> decorateConsumer(CircuitBreaker circuitBreaker, Consumer<T> consumer){
+        return decorateConsumer(circuitBreaker, consumer, CircuitBreakerResultPredicates.defaultPredicate1());
+    }
+
+    /**
+     * Returns a consumer which is decorated by a CircuitBreaker.
+
+     * @param circuitBreaker the CircuitBreaker
+     * @param consumer the original consumer
+     * @param resultPredicate result predicate by duration function.
+     * @param <T> the type of the input to the consumer
+     *
+     * @return a consumer which is decorated by a CircuitBreaker.
+     */
+    static <T> Consumer<T> decorateConsumer(CircuitBreaker circuitBreaker, Consumer<T> consumer, Function1<Long, Boolean> resultPredicate){
         return (t) -> {
             CircuitBreakerUtils.isCallPermitted(circuitBreaker);
             long start = System.nanoTime();
             try {
                 consumer.accept(t);
                 long durationInNanos = System.nanoTime() - start;
-                circuitBreaker.onSuccess(durationInNanos);
+
+                if (resultPredicate.apply(durationInNanos)) {
```

Comment:
@lespinsideg  the result predicate take the start time any reason why ? as u can simply do the calculation inside the if/else statement blocks .

Reply:
@Romeh Sorry but I didn't get your point. The result predicate doesn't take the start time. I thought duration could be a parameter for deciding the result.
https://github.com/resilience4j/resilience4j/issues/381